tools/src/main/java/org/apache/kafka/tools/JmxTool.java

```diff
@@ -0,0 +1,434 @@
+        }
+    }
+
+    private static void maybePrintDataRows(String reportFormat, Map<ObjectName, Integer> numExpectedAttributes, List<String> keys, Map<String, Object> attributes) {
+        if (attributes.size() == sumValues(numExpectedAttributes) + 1) {
+            switch (reportFormat) {
+                case "properties":
+                    keys.forEach(key -> System.out.println(String.format("%s=%s", key, attributes.get(key))));
+                    break;
+                case "csv":
+                    keys.forEach(key -> System.out.println(String.format("%s,\"%s\"", key, attributes.get(key))));
+                    break;
+                case "tsv":
+                    keys.forEach(key -> System.out.println(String.format("%s\t%s", key, attributes.get(key))));
+                    break;
+                default:
+                    System.out.println(mkString(keys.stream().map(attributes::get), ","));
+                    break;
+            }
+        }
+    }
+
+    private static class JmxToolOptions extends CommandDefaultOptions {
+        private final OptionSpec<String> objectNameOpt;
+        private final OptionSpec<String> attributesOpt;
+        private final OptionSpec<Integer> reportingIntervalOpt;
+        private final OptionSpec<Boolean> oneTimeOpt;
+        private final OptionSpec<String> dateFormatOpt;
+        private final OptionSpec<String> jmxServiceUrlOpt;
+        private final OptionSpec<String> reportFormatOpt;
+        private final OptionSpec<String> jmxAuthPropOpt;
+        private final OptionSpec<Boolean> jmxSslEnableOpt;
+        private final OptionSpec<Void> waitOpt;
+
+        public JmxToolOptions(String[] args) {
+            super(args);
+            objectNameOpt = parser.accepts("object-name", "A JMX object name to use as a query. This can contain wild cards, and this option " +
+                    "can be given multiple times to specify more than one query. If no objects are specified " +
+                    "all objects will be queried.")
+                .withRequiredArg()
+                .describedAs("name")
+                .ofType(String.class);
+            attributesOpt = parser.accepts("attributes", "The list of attributes to include in the query. This is a comma-separated list. If no " +
+                    "attributes are specified all objects will be queried.")
+                .withRequiredArg()
+                .describedAs("name")
+                .ofType(String.class);
+            reportingIntervalOpt = parser.accepts("reporting-interval", "Interval in MS with which to poll jmx stats; default value is 2 seconds. " +
+                    "Value of -1 equivalent to setting one-time to true")
+                .withRequiredArg()
+                .describedAs("ms")
+                .ofType(Integer.class)
+                .defaultsTo(2000);
+            oneTimeOpt = parser.accepts("one-time", "Flag to indicate run once only.")
+                .withOptionalArg()
```

Comment:
Previously this argument was required. Why is it optional now?

Reply:
Here I'm fixing the following issue with the old implementation:

It is rather common to pass boolean options without the argument (true is implicit).

```
$ bin/kafka-run-class.sh org.apache.kafka.tools.JmxTool \
  --jmx-url service:jmx:rmi:///jndi/rmi://:9990/jmxrmi \
  --object-name kafka.server:type=BrokerTopicMetrics,name=MessagesInPerSec \
  --attributes FifteenMinuteRate,FiveMinuteRate \
  --date-format "yyyyMMdd-hh:mm:ss" \
  --one-time \
  --report-format tsv
"time","kafka.server:type=BrokerTopicMetrics,name=MessagesInPerSec:FifteenMinuteRate","kafka.server:type=BrokerTopicMetrics,name=MessagesInPerSec:FiveMinuteRate"
20230201-09:39:41,0.0,0.0
```

As you can see, the parser sees `--report-format tsv` as the argument of `--one-time` and TSV formatting is ignored.

Setting optional args fixes the issue, without causing any breaking change.

```
$ bin/kafka-run-class.sh org.apache.kafka.tools.JmxTool   --jmx-url service:jmx:rmi:///jndi/rmi://:9990/jmxrmi \
  --object-name kafka.server:type=BrokerTopicMetrics,name=MessagesInPerSec \
  --attributes FifteenMinuteRate,FiveMinuteRate \
  --date-format "yyyyMMdd-hh:mm:ss" \
  --one-time \
  --report-format tsv
time    20230201-09:44:40
kafka.server:type=BrokerTopicMetrics,name=MessagesInPerSec:FifteenMinuteRate    0.0
kafka.server:type=BrokerTopicMetrics,name=MessagesInPerSec:FiveMinuteRate       0.0
```